docs/api/api-spec.md

```diff
@@ -126,12 +126,78 @@ It is expected that App Directories will also curate listed apps and ensure that
 
+## Context Sharing (DRAFT)
+
+On the financial desktop, applications can share related context and state with each other through the context sharing APIs. An application that wants to share context can simply publish a context object without being aware of how or by who the data is consumed.
+
+Applications can listen to all incoming context updates by declaring a generic context listener. If an application wants to be more selective and only wants to be notified of specific types of context, they can declare a typed context listener. The desktop agent will inspect the `type` property of context objects to route context updates only to interested applications. There is no limit to the amount of context listeners a receiving application can declare.
+
+When declaring a listener, a handle is returned that applications can use to unsubscribe from future updates.
+
+Applications that would like to query the current state of the desktop when they start up, can ask the desktop agent for the current value of a particular context type. This will return, asynchronously, the most recent value of that type, or no value, if none has been published for that type. An example of this behaviour would be a charting application that joins a group of trading applications, and would like to initialise itself to the currently selected instrument on the desktop.
+
+### Setting context
+
+Applications use the broadcast operation to publish context data objects to the desktop.
+
+```ts
+const instrument = {
+    type: 'fdc3.instrument',
+    id: {
+        ticker: 'AAPL';
+    }
+};
+
+desktopAgent.broadcast(instrument);
+```
+
+> Proposal: Can we change the name? e.g. `shareContext`/`setContext`.
+>
+> Reason: Broadcast doesn't imply the saving of the value for later `getCurrentValue` call.
+
+### Getting context
+
+If an application declares a generic context listener, providing a callback, they will be notified whenever another application invokes the `broadcast` function for any object:
+
+```ts
+const listener = desktopAgent.addContextListener(context => {
+    // handle context object
+});
+```
+
+If an application declares a typed context listener, the callback will only be invoked for context object updates where the type matches:
+
+```ts
+const instrumentListener = desktopAgent.addTypedContextListener('fdc3.instrument', instrument => {
+    // handle instrument object
+});
+```
+
+If an application would like to retrieve the current state of the desktop at startup, they can ask for the current value for a particular type of object:
+
+```ts
+await instrument = desktopAgent.getCurrentValue('fdc3.instrument');
```

Comment:
`getCurrentValue` doesn't really tell you much about what this function does. I'd suggest `getContext` or `getContextValue` - the fact that it is current is implicit.